tests/types.test.tsx

```diff
@@ -1,8 +1,27 @@
 import { expectType } from 'ts-expect'
-import { atom, useAtom } from 'jotai'
 import type { Atom, PrimitiveAtom, WritableAtom } from 'jotai'
+import { atom, useAtom } from 'jotai'
+import { useAtomValue, waitForAll } from 'jotai/utils'
 import type { SetAtom } from '../src/core/atom'
 
+it('waitForAll() should return the correct types', () => {
+  function Component() {
+    const numberAtom = atom(async () => 0)
+    const stringAtom = atom(async () => '')
+
+    const [number, string] = useAtomValue(waitForAll([numberAtom, stringAtom]))
+    expectType<number>(number)
+    expectType<string>(string)
+
+    const { numberAtom: number2, stringAtom: string2 } = useAtomValue(
+      waitForAll({ numberAtom, stringAtom })
+    )
+    expectType<number>(number2)
+    expectType<string>(string2)
+  }
+  Component
+})
```

Comment:
We are still not sure, but I feel like we should put this in `tests/utils`.
Can you make a new `tests/utils/waitForAll.types.text.tsx` file meanwhile?

Reply:
👍